programs/drift/src/controller/orders.rs

```diff
@@ -292,41 +292,28 @@ pub fn place_perp_order(
     let worst_case_base_asset_amount_after =
         user.perp_positions[position_index].worst_case_base_asset_amount()?;
 
-    let position_base_asset_amount = user.perp_positions[position_index].base_asset_amount;
-    let base_asset_amount_unfilled = user.orders[new_order_index]
-        .get_base_asset_amount_unfilled(Some(position_base_asset_amount))?;
-    let order_risk_decreasing = is_order_risk_decreasing(
-        &params.direction,
-        base_asset_amount_unfilled,
-        position_base_asset_amount,
-    )?;
+    let risk_increasing = worst_case_base_asset_amount_after.unsigned_abs()
+        > worst_case_base_asset_amount_before.unsigned_abs();
 
-    let risk_decreasing = worst_case_base_asset_amount_after.unsigned_abs()
-        <= worst_case_base_asset_amount_before.unsigned_abs()
-        && order_risk_decreasing;
+    options.update_risk_increasing(risk_increasing);
 
     // when orders are placed in bulk, only need to check margin on last place
     if options.enforce_margin_check {
-        // Order fails if it's risk increasing and it brings the user collateral below the margin requirement
-        let meets_initial_margin_requirement = meets_place_order_margin_requirement(
+        meets_place_order_margin_requirement(
```

Comment:
previously skipped if order wasn't risk increasing. now it will check maintenance if not risk increasing